src/vmm/src/devices/legacy/i8042.rs

```diff
@@ -119,18 +119,18 @@ pub struct I8042Device {
 
 impl I8042Device {
     /// Constructs an i8042 device that will signal the given event when the guest requests it.
-    pub fn new(reset_evt: EventFd, kbd_interrupt_evt: EventFd) -> I8042Device {
-        I8042Device {
+    pub fn new(reset_evt: EventFd) -> Result<I8042Device, std::io::Error> {
+        Ok(I8042Device {
             reset_evt,
-            kbd_interrupt_evt,
+            kbd_interrupt_evt: EventFd::new(libc::EFD_NONBLOCK)?,
```

Comment:
just assert on eventfd

Reply:
Yes, and no. It is in the `InstanceStart` API path, but if there is an error anywhere in VM creation, FC [exits](https://github.com/firecracker-microvm/firecracker/blob/main/src/firecracker/src/main.rs#L429), so asserting here does not change the behavior, but saves us some error propagation sanity.

Reply:
I thought that we were sending an Error response to the client before exiting. :thinking: